google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -120,6 +121,15 @@ public CancellableRunnable createDeployRunner(
       throw new RuntimeException("Invalid deployment source selected for deployment");
     }
 
+    if (CloudSdkService.getInstance().validateCloudSdk().contains(
+        CloudSdkValidationResult.CLOUD_SDK_NOT_FOUND)) {
+      callback.errorOccurred(GctBundle.message("appengine.cloudsdk.location.invalid.message") + " "
+          + CloudSdkService.getInstance().getSdkHomePath());
+      logger.warn(GctBundle.message("appengine.cloudsdk.location.invalid.message") + " "
```

Comment:
There is a second check which also returns null, like this one.

The error message is getting correctly printed and the deployment correctly aborted, so my goal is done without having to throw an exception. I don't know why the first check throws an exception, though, but I also wonder if we really need to there.